File: src/main/java/com/chaottic/nbt/number/ShortTag.java
package com.chaottic.nbt.number;

import java.io.DataInput;
import java.io.IOException;

public record ShortTag(short s) implements NumberTag {
    @Override
    public byte getAsByte() {
        return (byte) s;
    }

    @Override
    public short getAsShort() {
        return s;
    }

    @Override
    public int getAsInt() {
        return s;
    }

    @Override
    public long getAsLong() {
        return s;
    }

    @Override
    public float getAsFloat() {
        return s;
    }

    @Override
    public double getAsDouble() {
        return s;
    }

    public static ShortTag createShortTag(DataInput dataInput) throws IOException {
        return new ShortTag(dataInput.readShort());
    }
}

File: src/main/kotlin/com/chaottic/nbt/number/ShortTag.kt
package com.chaottic.nbt.number

import com.chaottic.nbt.TagType
import java.io.DataInput

data class ShortTag(private val s: Short) : NumberTag {

	override fun asByte() = s.toByte()
	override fun asShort() = s
	override fun asInt() = s.toInt()
	override fun asLong() = s.toLong()
	override fun asFloat() = s.toFloat()
	override fun asDouble() = s.toDouble()

	companion object : TagType<ShortTag> {

		override fun createTag(dataInput: DataInput) = ShortTag(dataInput.readShort())
	}
}
